lib/dependencies.js

```diff
@@ -52,6 +53,13 @@ module.exports = class Dependencies {
      * @type Map<string, Array<string>>;
      */
     this.dependentsMap = new Map();
+    /**
+     * Custom fs object can be passed to the custructor.
+     * This helps us to pass the this.input of the broccoli-plugin
+     * to keep the encapsulations.
+     * @type {typeof fs}
+     */
+    this.fs = options.fs || fs;
```

Comment:
Although functionally correct, this fallback should most likely be removed. Under what circumstances should we allow native `fs` to be used now?

My feeling is that, allowing for this fallback may hide bugs. Specifically bugs where some future PR accidentally causes `options.fs` to be falsy causing an erroneous fallback to `fs`. The resulting issues would likely be hard to diagnose.

Since this is an internal private object, I would prefer if we make it exactly what we need, and make options.fs mandatory, with no graceful fallback

Reply:
This was done to provide backward compatibility for those who extended the `Filter` class.

Reply:
I am not sure if there is a real use case for this. 
an example will be
```js
class awk extends Filter {
build () {
// we can won't break this implementation.
let srcDir = this.inputPaths[0];
 this.dependencies = this.processor.initialDependencies(srcDir);
 // .... do build related stuffs.
}
processString(content) {
 return content;
}
}

The current implementation need to make a slight change where I assume srcDirs in `new Dependency(srcDirs)` always receives array, I have to make a change to convert if the argument is not an array change to an array.

Reply:
@SparshithNR that would suggest to me, that we have just changed the signature of `initializeDependencies` in a breaking way, that is unexpected. Users having to remember to pass `this.input` if they invoke the `initialDependencies` stuff seems very unfortunate.

Can we pass the needed details to the processor when it is initialized? That would ensure no breaking changes occurs.

Reply:
Unfortunately no, we cannot pass it while initializing the processor. The processor gets initialized at the constructor and `this.input` is accessible only at the `build` time.

Reply:
We can probably add a function in the processor, `setcustomFS` and set `this.customFS` on `this.processor` or `set it directly in the `build` method. 
```js
build() {
this.processor.customFS = this.input;
```
or 
```js
build() {
this.processor.setCustomFS(this.input);
```